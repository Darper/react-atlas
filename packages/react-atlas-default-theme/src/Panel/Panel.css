.accordion_panel .panelContainer {
    display: none;
    padding-left: 0rem;
    padding-right: 0rem;
}