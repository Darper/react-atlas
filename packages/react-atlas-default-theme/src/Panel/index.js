import * as PanelStyle from "./panel.css";
export { PanelStyle };
