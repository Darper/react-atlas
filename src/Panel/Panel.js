import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Panel.css";

export class Panel extends React.PureComponent {
  constructor(props) {
    super(props);
  }
  render() {
    const panelStyles = cx({
      "panel": true
    });
    const { className, children, style } = this.props;
    return (
      <div style={style} className={cx(className)} styleName={panelStyles}>
        {children}
      </div>
    );
  }
}

Panel.propTypes = {
  /**
   * Text, any HTML element, or React Component.
   */
  "children": PropTypes.node,

  /** An object, array, or string of CSS classes to apply to Panel.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * Pass inline styling here.
   */
  "style": PropTypes.object
};

export default CSSModules(Panel, styles, { "allowMultiple": true });
