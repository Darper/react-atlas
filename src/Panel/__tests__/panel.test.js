import React from "react";
import { PanelCore } from "../../../react-atlas-core/src/Panel/index";

import renderer from "react-test-renderer";

describe("Test correct render", () => {
  it("Test correct render", function() {
    const tree = renderer.create(<PanelCore>Panel Text</PanelCore>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
