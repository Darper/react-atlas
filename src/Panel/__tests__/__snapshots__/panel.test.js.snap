// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test correct render Test correct render 1`] = `
<div
  className="panel"
  style={undefined}
>
  Panel Text
</div>
`;
